### PR TITLE
Add cpu_requests query Closes #487

### LINKED ISSUE #487: Add cpu_requests query for Kubernetes
### Platform
Kubernetes

### Description
Add new "CPU Requests" query, that checks if CPU requests are set

### PR DESCRIPTION
**Pull request by rafaela-soares**
Since attribute 'containers' is an array, it is necessary to check if all containers have CPU requests.

Closes #487

### CODE REVIEW

File: assets/queries/k8s/cpu_requests/query.rego [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package Cx
```
Comment (joelsou5a) [inline assets/queries/k8s/cpu_requests/query.rego:1]: Check the query attached, it does have a more specific searchKey

[query.zip](https://github.com/Checkmarx/kics/files/5629863/query.zip)

File: assets/queries/k8s/cpu_requests/test/positive_expected_result.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+[
+	{
+		"queryName": "CPU Requests",
+		"severity": "MEDIUM",
+		"line": 6
```
Comment (joelsou5a) [inline assets/queries/k8s/cpu_requests/test/positive_expected_result.json:5]: "line": 11

**Review by joelsou5a — CHANGES_REQUESTED** [review]

File: assets/queries/k8s/cpu_requests/query.rego [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+CxPolicy [ result ] {
+    document := input.document[i]
+    specInfo := getSpecInfo(document)
+    metadata := document.metadata
+    
+    containers := document.spec.containers
+    
+    object.get(containers[index]["resources"]["requests"], "cpu", "undefined") == "undefined"
```
Comment (joaomartinscx) [inline assets/queries/k8s/cpu_requests/query.rego:10]: In this case, when resources or requests aren't present, the query will not detect these edge cases. I suggest to test them separately.

**Review by joaomartinscx — CHANGES_REQUESTED** [review]

File: assets/queries/k8s/cpu_requests/query.rego [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    
+    
+    result := {
+                "documentId":        input.document[i].id,
+                "issueType":         "MissingAttribute",
+                "searchKey":         sprintf("metadata.name=%s.%s.containers.name=%s.resources.requests", [metadata.name, specInfo.path, containers[index].name]),
+                "keyExpectedValue":  sprintf("%s.containers.name=%s does have CPU requests",[specInfo.path, containers[index].name]),
+                "keyActualValue":    sprintf("%s.containers.name=%s doesn't have CPU requests",[specInfo.path, containers[index].name])
```
Comment (joaomartinscx) [inline assets/queries/k8s/cpu_requests/query.rego:18]: Make clear the path where the content should be present

**Review by joaomartinscx — CHANGES_REQUESTED** [review]

**Review by joaomartinscx — COMMENTED** [review]
Everything looks good to me, I recommend approval and merge of this PR.

**Review by joelsou5a — APPROVED** [review]
